crowdin/crowdin.go

```diff
@@ -0,0 +1,235 @@
+package crowdin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+const (
+	baseURL    = "https://api.crowdin.com/"
+	apiVersion = "api/v2"
+
+	userAgent = "crowdin-api-client-go/0.0.1"
+)
+
+// Client is a Crowdin API client.
+type Client struct {
+	baseURL    *url.URL
+	token      string
+	userAgent  string
+	httpClient *http.Client
+}
+
+// EnterpriseClient is a Crowdin Enterprise API client.
+type EnterpriseClient struct {
+	Client
+}
+
+// NewClient creates a new Crowdin API client with provided options (ex. WithHTTPClient).
+// `token` is a personal access token.
+func NewClient(token string, opts ...ClientOption) (*Client, error) {
```

Comment:
Let's combine the Crowdin and Enterprise clients together. If the optional organization is passed - just build the base URL based on the passed value and override the default one. There is no need to have separate implementations for this.